ADAL/src/ADAuthenticationResult+Internal.m

```diff
@@ -155,20 +158,48 @@ + (ADAuthenticationResult*)resultForBrokerErrorResponse:(NSDictionary*)response
     
     // Otherwise parse out the error condition
     ADAuthenticationError* error = nil;
+    NSMutableDictionary* userInfo = nil;
     
     NSString* errorDetails = [response valueForKey:OAUTH2_ERROR_DESCRIPTION];
     if (!errorDetails)
     {
         errorDetails = @"Broker did not provide any details";
     }
+
+    if (response[BROKER_APP_VERSION])
+    {
+        [userInfo setValue:response[BROKER_APP_VERSION] forKey:BROKER_APP_VERSION];
+    }
         
     NSString* strErrorCode = [response valueForKey:@"error_code"];
     NSInteger errorCode = AD_ERROR_TOKENBROKER_UNKNOWN;
     if (strErrorCode && ![strErrorCode isEqualToString:@"0"])
     {
         errorCode = [strErrorCode integerValue];
     }
+
     
+
+    if (errorCode == AD_ERROR_SERVER_PROTECTION_POLICY_REQUIRED)
+    {
+        // For protection_policy_required error, add extra info for the app in the userInfo dictionary of the error
+        userInfo = [[NSMutableDictionary alloc] initWithCapacity:3];
+        if (response[@"suberror"])
+        {
+            [userInfo setValue:response[@"suberror"] forKey:@"suberror"];
+        }
+
+        if (response[@"userId"])
+        {
+            [userInfo setValue:response[@"userId"] forKey:@"userId"];
+        }
+
+        if (response[@"intuneTokenError"])
```

Comment:
@jasoncoolmax what is intuneTokenError?

Reply:
I think this is mistakenly added when Mike addressed the comments, Mike can confirm :)

Reply:
I originally attached the error from decrypting the Intune token if one was generated, but on further thought this isn't useful to the developer as this optimization should be transparent, I've removed it.